editor/src/components/navigator/left-pane/index.tsx

```diff
@@ -74,13 +75,22 @@ export const LeftPaneComponent = React.memo<LeftPaneComponentProps>((props) => {
 
   const isMyProject = useIsMyProject()
 
+  const curriedRequireFn = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.codeResultCache.curriedRequireFn,
+    'LeftPaneComponent curriedRequireFn',
+  )
+
   const isRemixProject = useEditorState(
     Substores.projectContents,
-    (store) =>
-      getRemixRootFile(
-        getRemixRootDir(store.editor.projectContents),
-        store.editor.projectContents,
-      ) != null,
+    (store) => {
+      const rootDir = getRemixRootDir(store.editor.projectContents, curriedRequireFn)
```

Comment:
why not just use `store.editor.curriedRequireFn` here? and then you could delete the `const curriedRequireFn = useEditorState...` line

Reply:
Because it's not part of that substore?

Reply:
ah right. boo! how frequently does the curriedRequireFn change? it may be beneficial to change the substore type so we do not rerender the left-pane every time curriedRequireFn changes

Reply:
We could conceivably disconnect it from changes to `curriedRequireFn`, but that definitely makes me a little antsy around it not updating when it might need to.